src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const modifyState = { ...state };
+
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const entry of Object.entries(action.extraData)) {
+          modifyState[entry[0]] = entry[1];
+        }
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete modifyState[key];
+        }
+        break;
+      case 'clear':
+        for (const key of Object.keys(modifyState)) {
+          delete modifyState[key];
+        }
+        break;
+    }
```

Comment:
The 'switch' statement is missing a 'default' case. It's a good practice to include a 'default' case for error handling, even if it's just to throw an error or to continue without action.